src/qibolab/platforms/abstract.py

```diff
@@ -323,81 +379,40 @@ def __call__(self, sequence, nshots=None):
         return self.execute_pulse_sequence(sequence, nshots)
 
     def create_RX90_pulse(self, qubit, start=0, relative_phase=0):
-        qd_duration = self.settings["native_gates"]["single_qubit"][qubit]["RX"]["duration"]
-        qd_frequency = self.settings["native_gates"]["single_qubit"][qubit]["RX"]["frequency"]
-        qd_amplitude = self.settings["native_gates"]["single_qubit"][qubit]["RX"]["amplitude"] / 2
-        qd_shape = self.settings["native_gates"]["single_qubit"][qubit]["RX"]["shape"]
-        qd_channel = self.settings["qubit_channel_map"][qubit][1]
-        from qibolab.pulses import Pulse
-
-        return Pulse(start, qd_duration, qd_amplitude, qd_frequency, relative_phase, qd_shape, qd_channel, qubit=qubit)
+        kwargs = self.qubits[qubit].get_native_gate("RX", start, relative_phase)
```

Comment:
I do not see the benefit of this refactor. Here, the platform is generating the Pulse to implement a gate, and for that it has to ask the qubit object for data, that it already had.
This refactor seems to be sending data from the platform to qubit then to be brought back to platform.